libraries/AP_Scripting/generator/src/main.c

```diff
@@ -273,12 +274,29 @@ void handle_header(void) {
     error(ERROR_HEADER, "Header %s was already included on line %d", name, node->line);
   }
 
+  char * key_word = next_token();
+  char *depends = NULL;
+  if (key_word != NULL) {
+    if (strcmp(key_word, keyword_depends) != 0) {
+      error(ERROR_HEADER, "Headers only support depends keyword, got %s", key_word);
+    }
+    depends = strtok(NULL, "");
+    if (depends == NULL) {
+      error(ERROR_HEADER, "Expected a depends string for %s", name);
+    }
+  }
+
   // add to the list of headers
   node = (struct header *)allocate(sizeof(struct header));
   node->next = headers;
   node->line = state.line_num;
   node->name = (char *)allocate(strlen(name) + 1);
   strcpy(node->name, name);
+  if (depends != NULL) {
+    node->dependency = (char *)allocate(strlen(depends) + 1);
+    strcpy(node->dependency, depends);
```

Comment:
```suggestion
    string_copy(&(node->dependency), depends);
```

We have a helper to wrap this up and be clearer.

Reply:
fixed, I had to move `string_copy` up in the file to get it to work